sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcSourceSuite.scala

```diff
@@ -162,6 +162,40 @@ abstract class OrcSuite extends QueryTest with TestHiveSingleton with BeforeAndA
       hiveClient.runSqlHive("DROP TABLE IF EXISTS orc_varchar")
     }
   }
+
+  test("SPARK-19459: read char/varchar column written by Hive") {
+    val hiveClient = spark.sharedState.externalCatalog.asInstanceOf[HiveExternalCatalog].client
+    val location = Utils.createTempDir().toURI
+    try {
+      hiveClient.runSqlHive(
+        """
+           |CREATE EXTERNAL TABLE hive_orc(
+           |  a STRING,
+           |  b CHAR(10),
+           |  c VARCHAR(10))
+           |STORED AS orc""".stripMargin)
+      // Hive throws an exception if I assign the location in the create table statment.
+      hiveClient.runSqlHive(
+        s"ALTER TABLE hive_orc SET LOCATION '$location'")
+      hiveClient.runSqlHive(
+        "INSERT INTO TABLE hive_orc SELECT 'a', 'b', 'c' FROM (SELECT 1) t")
+
```

Comment:
How about adding one more check?

```Scala
      checkAnswer(spark.table("hive_orc"), Row("a", "b         ", "c"))
```

Then, we can remove the test case `SPARK-18220: read Hive orc table with varchar column`